shared/js/background/utils.es6.js

```diff
@@ -114,6 +140,22 @@ function getAsyncBlockingSupport () {
     return false
 }
 
+// return true if the given url is in the safelist. For checking if the current tab is in the safelist,
+// tabManager.site.whitelisted is the preferred method.
+function isSafeListed (url) {
+    const hostname = extractHostFromURL(url)
+    const tld = tldts.parse(url).domain
+    const safeList = settings.getSetting('whitelisted')
+    let subdomains = hostname.split('.')
+    while (subdomains.length > 1) {
+        if (safeList && safeList[subdomains.join('.')]) {
+            return true
+        }
+        subdomains.shift()
+    }
+    return safeList && (safeList[hostname] || safeList[tld])
```

Comment:
so now that you have that `while` loop, I think we can change this last line to `return false;`? In fact, you probably don't need `tldts.parse` anymore either.